core/constraints/constraints.go

```diff
@@ -181,16 +185,33 @@ func (v *Value) HasInstanceType() bool {
 	return v.InstanceType != nil && *v.InstanceType != ""
 }
 
+// AddSpace is responsible for adding a space constraint to [Value]. If [Value]
+// currently has a nil slice of spaces a new slice will be allocated. If the
+// space to be added is considered "excluded" then an exclude prefix will be
+// applied to the name of the constraint.
+func (v *Value) AddSpace(space string, excluded bool) {
```

Comment:
Do we want to implement a merge instead of append?

Reply:
I am inclined not to. This would be a break from our current convention and logic. I don't want to change the schematics until we have a reason :)